.github/workflows/basePipeline.yml

```diff
@@ -0,0 +1,65 @@
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 23
+        uses: actions/setup-java@v4
+        with:
+          java-version: '23.0.0+37'
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@af1da67850ed9a4cedd57bfd976089dd991e2582 # v4.0.0
+
+      - name: Check directory
+        run:
+          ls
+
+      - name: Grant Permissions to gradlew
+        run:
+          cd demo;
```

Comment:
Is 'demo' a placeholder name, shouldn't this be the name of the project folder? I believe it's 'project'